### PR TITLE
Update Secret Manager (pre-release)

### PR DESCRIPTION
**Pull request by blowmage**
* Update generated files
* Update veneer Authentication Guide
* Update veneer factory method

### CODE REVIEW

File: google-cloud-secret_manager/lib/google/cloud/secret_manager.rb [hunk 1]
```diff
@@ -18,26 +18,26 @@
...
+    # Secret Manager provides a secure and convenient tool for storing API keys, passwords, certificates, and other
+    # sensitive data.
     module SecretManager
       ##
-      # TODO
-      def self.new version: :v1beta1, service: :secret_manager_service, &block
+      # Create a new `SecretManagerService::Client` object.
+      #
+      # @param version [String, Symbol] The API version to create the client instance. Optional. If not provided
+      #   defaults to `V1beta1`.
```
Comment (dazuma) [inline google-cloud-secret_manager/lib/google/cloud/secret_manager.rb]: defaults to `:v1beta1`

File: google-cloud-secret_manager/test/google/cloud/secret_manager/version_test.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2015 Google LLC
+# Copyright 2019 Google LLC
```
Comment (dazuma) [inline google-cloud-secret_manager/test/google/cloud/secret_manager/version_test.rb]: 2020

File: google-cloud-secret_manager/lib/google/cloud/secret_manager.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-secret_manager/lib/google/cloud/secret_manager.rb]: Fixed.

File: google-cloud-secret_manager/test/google/cloud/secret_manager/version_test.rb (hunk 2 above)
Comment (blowmage) [reply google-cloud-secret_manager/test/google/cloud/secret_manager/version_test.rb]: Fixed.

**Pull request merged**